### PR TITLE
refactor: Extract common image functionnalities

### PR DESCRIPTION
**Pull request by atomrc**
## Description

This will harmonize how images are loaded across the app (using the same component (namely `Image`) to load images everywhere)

This should fix quite a few bugs related to images jumping in size when switching states (from loading to loaded for example). 
It should also quite improve scrolling up in a conversation and avoid random jumps

### Scrolling up in the conversation with images (no jump)

https://github.com/wireapp/wire-webapp/assets/1090716/ab8046cf-ff3b-4056-8320-e134e0f1c8fa

### Sending/loading and replying to images

https://github.com/wireapp/wire-webapp/assets/1090716/452bfb3a-2f72-4a8d-bfe2-2ce8f0f3cb4a

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/Image.tsx [hunk 1]
```diff
@@ -17,83 +17,133 @@
...
+          }
+          setImageUrl(url);
+        } catch (error) {
+          console.error(error);
         }
-        setAssetIsLoading(false);
-      });
+      })();
     }
+  }, [imageUrl, isInViewport, image, isFileSharingReceivingEnabled, getAssetUrl]);
```
Comment (atomrc) [inline src/script/components/Image.tsx]: This entire loading logic now only exist in a single place: here. 

Previously it was also duplicated in the `ImageAsset` component

File: src/script/components/MessagesList/Message/ContentMessage/asset/ImageAsset/ImageAsset.tsx [hunk 2]
```diff
@@ -17,84 +17,31 @@
...
-    }
-  }, [imageUrl, isInViewport, resource, isFileSharingReceivingEnabled, getAssetUrl]);
-
-  useEffect(() => {
-    return () => {
-      isUnmounted.current = true;
-      imageUrl?.dispose();
-    };
-  }, []);
-
```
Comment (atomrc) [inline src/script/components/MessagesList/Message/ContentMessage/asset/ImageAsset/ImageAsset.tsx:96]: This entire logic is now in the `Image` component

File: src/script/components/MessagesList/Message/ContentMessage/asset/ImageAsset/ImageAsset.tsx [hunk 3]
```diff
@@ -103,70 +50,37 @@ export const ImageAsset = ({
...
-        height: asset.height,
-      }),
-  };
-
-  const imageStyle: CSSObject = {
-    width: imageWidth,
-    maxWidth: '100%',
-    height: 'auto',
+    width: asset.width,
+    aspectRatio: asset.ratio,
```
Comment (atomrc) [inline src/script/components/MessagesList/Message/ContentMessage/asset/ImageAsset/ImageAsset.tsx:55]: This image container now has a rather simple rendering logic:
It tries to match the image size and stay contained in the height and width contraints we give it

File: src/script/components/Image.tsx [hunk 4]
```diff
@@ -17,83 +17,131 @@
...
 
-export interface ImageProps extends React.HTMLProps<HTMLDivElement> {
-  aspectRatio?: number;
-  width?: string;
-  asset: AssetRemoteData;
-  assetRepository?: AssetRepository;
-  click?: (asset: AssetRemoteData, event: React.MouseEvent) => void;
+const imageWrapperStyle = {
+  width: '100%',
+};
```
Comment (V-Gira) [inline src/script/components/Image.tsx]: I don't feel strongly about it, but I think it's missing
```javascript
  backgroundColor: 'var(--foreground-fade-8)',
  ```
from the previous imlementation, no?

Comment (atomrc) [reply src/script/components/Image.tsx]: It was actually just for the loading dots. They now have this style just for them  (see https://github.com/wireapp/wire-webapp/pull/16810/files#diff-e36270aa90cae085e55b2fa24b4aba8e59a4d0455b87235a77c611aa8bd5883aR25)

**Review by PatrykBuniX — APPROVED** [review]
Looks solid 👌

**Review by V-Gira — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**